src/main/java/org/snomed/aag/data/services/CriteriaItemService.java

```diff
@@ -165,4 +170,54 @@ public void verifyManual(CriteriaItem criteriaItem, boolean expectedManual) {
 			throw new AccessDeniedException("Criteria Item cannot be changed manually.");
 		}
 	}
+
+	/**
+	 * Remove entry from collection. If entry does not have a corresponding flag in Branch metadata
+	 * or the corresponding flag in Branch metadata is false, then remove entry from collection.
+	 *
+	 * @param criteriaItems Collection to process
+	 * @param branch        Branch to cross reference
+	 */
+	public void removeNonEnabled(Set<CriteriaItem> criteriaItems, Branch branch) {
+		verifyParams(criteriaItems, branch);
+		Set<String> authorFlags = MetadataUtil.getEnabledAuthorFlags(branch).keySet();
+		if (!authorFlags.isEmpty()) {
+			criteriaItems.removeIf(item -> item.getEnabledByFlag().stream().noneMatch(authorFlags::contains));
+		}
+	}
+
+	/**
+	 * Remove entry from collection. If there is a corresponding flag in Branch metadata,
+	 * remove entry if the flag is disabled. If there is no corresponding flag in Branch metadata,
+	 * entry is not removed to prevent a lack of metadata overwriting system.
+	 *
+	 * @param criteriaItems Collection to process
+	 * @param branch        Branch to cross reference
+	 */
+	public void removeExemptFromCompletionVerification(Set<CriteriaItem> criteriaItems, Branch branch) {
+		Map<String, Object> authorFlags = MetadataUtil.getAllAuthorFlags(branch);
+		if (!authorFlags.isEmpty()) {
+			Set<Map.Entry<String, Object>> authorFlagEntries = authorFlags.entrySet();
+			criteriaItems.removeIf(item -> {
+				Set<String> itemFlags = item.getEnabledByFlag();
+				boolean remove = true;
+				for (Map.Entry<String, Object> entry : authorFlagEntries) {
+					boolean hasFlag = itemFlags.contains(entry.getKey());
+					if (!hasFlag) {
+						// Prevent metadata overwriting PAC when metadata has no corresponding flag
+						return false;
+					}
+
+					boolean flagEnabled = Boolean.parseBoolean(entry.getValue().toString());
+					if (flagEnabled) {
+						// Item is enabled in metadata, therefore, keep and validate
+						remove = false;
+						break;
+					}
+				}
+
+				return remove;
+			});
+		}
+	}
```

Comment:
This method was added following your suggestion of adding the author flag check to the promotion workflow. 

It's a tad confusing... hopefully my comments etc have explained it enough. Essentially, if the author flag is present, take the CriteriaItem through to completion verification _only_ if it is true. Otherwise (both negative & null) ignore it.

Reply:
I'm struggling to understand why you could not use `removeNonEnabled` rather than this method?

Reply:
For the record: Kai and I had a call. 

I have removed the `removeExemptFromCompletionVerification` method and instead I am using the other `removeNonEnabled` method. 

The `removeNonEnabled` has subsequently been renamed to `removeNonEnabledMarkMandatory` as it now marks an item as mandatory if the mandatory flag has been found.